### PR TITLE
Issue 6274: keeping all backing votes in provisioner vote set

### PR DESCRIPTION
**Pull request by BradleyOlson64**

### CODE REVIEW

**Comment by ordian** [comment]
The problem with always importing backing statements is that runtime dispute filtering code will discard any sets with duplicate votes. So we need to modify the runtime first to allow passing duplicate backing statements I guess?

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2214636

**Comment by BradleyOlson64** [comment]
> The problem with always importing backing statements is that runtime dispute filtering code will discard any sets with duplicate votes. So we need to modify the runtime first to allow passing duplicate backing statements I guess?

So perhaps I should create a separate pr containing the runtime changes and get that merged first?

**Comment by ordian** [comment]
Upon further inspection, it seems my initial comment was wrong:
https://github.com/paritytech/polkadot/blob/6a5d31fcbdd3ac5dd880e0bf3616e303b5bb480f/runtime/parachains/src/disputes.rs#L939

**Review by ordian — APPROVED** [review]

**Review by tdimitrov — APPROVED** [review]

**Comment by tdimitrov** [comment]
Good job! Check the tests - `resume_dispute_without_local_statement` is failing.

**Comment by BradleyOlson64** [comment]
bot merge

File: node/core/provisioner/src/disputes/prioritized_selection/mod.rs [hunk 1]
```diff
@@ -364,10 +364,20 @@ fn is_vote_worth_to_keep(
...
 	};
+	// We want to keep all backing votes. This maximizes the number of backers
+	// punished when misbehaving.
+	if let Some(kind) = valid_kind {
+		match kind {
+			ValidDisputeStatementKind::BackingValid(_) => return true,
+			ValidDisputeStatementKind::BackingSeconded(_) => return true,
+			_ => (),
+		}
+	}
```
Comment (mrcnski) [inline node/core/provisioner/src/disputes/prioritized_selection/mod.rs:379]: nit: Your version is totally fine, but I think you can also do something like this 🙂 (haven't checked it):

```suggestion
    if matches!(valid_kind, Some(ValidDisputeStatementKind::BackingValid(_) | ValidDisputeStatementKind::BackingSeconded(_)) {
		return true
	}
```

**Review by mrcnski — APPROVED** [review]
Make sure you add `Closes #6274` to the description, so the issue gets closed automatically.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2218525

**Comment by BradleyOlson64** [comment]
> Good job! Check the tests - `resume_dispute_without_local_statement` is failing.

Could use a second pair of eyes on this. It seems like the chain scraper is asking for block ancestors when we don't expect it to. 

Thing is, for the life of me I can't figure out how keeping backing votes in the provisioner effects this at all.

The failing test doesn't even include any backing votes, so it seems like my changes wouldn't come into play.

**Comment by BradleyOlson64** [comment]
bot merge

**Pull request merged**